### PR TITLE
feat: add model script, training recipe and pretrained weight of cmt_s

### PR DESCRIPTION
**Pull request by wcrzlh**
Thank you for your contribution to the MindCV repo.
Before submitting this PR, please make sure:

- [x] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [ ] You have added unit tests

## Motivation

The model script, training recipe and pertained weight of cmt_small is added.

## Test Plan

Please refer to 'configs/cmt/README.md' for testing and reproducing.

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

File: mindcv/models/cmt.py [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+
+default_cfgs = {
+    "cmt_small": _cfg(url='https://download.mindspore.cn/toolkits/mindcv/cmt/cmt_small-6858ee22.ckpt')
+}
+
+
+to_2tuple = _ntuple(2)
+
+
+class DropPath(nn.Cell):
```
Comment (geniuspatrick) [inline mindcv/models/cmt.py:37]: 这个可以用公共层的实现吗

File: mindcv/models/cmt.py [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+            random_tensor = self.rand((x_shape[0], 1, 1))
+            random_tensor = random_tensor + self.keep_prob
+            random_tensor = self.floor(random_tensor)
+            x = x / self.keep_prob
+            x = x * random_tensor
+        return x
+
+
+def swish(x):
+    return x * P.Sigmoid()(x)
```
Comment (geniuspatrick) [inline mindcv/models/cmt.py:62]: 函数式接口ops.sigmoid

**Review by geniuspatrick — CHANGES_REQUESTED** [review]

**Comment by wcrzlh** [comment]
fixed

File: mindcv/models/cmt.py [hunk 3]
```diff
@@ -0,0 +1,451 @@
...
+        x = ops.transpose(x.reshape(_B, _C, _H * _W), (0, 2, 1))
+        x = self.norm(x)
+
+        H, W = H // self.patch_size[0], W // self.patch_size[1]
+        return x, (H, W)
+
+
+class CMT(nn.Cell):
+    def __init__(
+            self,
```
Comment (geniuspatrick) [inline mindcv/models/cmt.py:207]: 这里缩进降低4个空格

File: mindcv/models/cmt.py [hunk 4]
```diff
@@ -0,0 +1,451 @@
...
+        self.patch_embed_a = PatchEmbed(
+            img_size=img_size // 2, patch_size=2, in_chans=stem_channel, embed_dim=embed_dims[0])
+        self.patch_embed_b = PatchEmbed(
+            img_size=img_size // 4, patch_size=2, in_chans=embed_dims[0], embed_dim=embed_dims[1])
+        self.patch_embed_c = PatchEmbed(
+            img_size=img_size // 8, patch_size=2, in_chans=embed_dims[1], embed_dim=embed_dims[2])
+        self.patch_embed_d = PatchEmbed(
+            img_size=img_size // 16, patch_size=2, in_chans=embed_dims[2], embed_dim=embed_dims[3])
+
+        self.relative_pos_a = ops.Zeros()(
```
Comment (geniuspatrick) [inline mindcv/models/cmt.py:255]: ops.zeros()

**Review by geniuspatrick — CHANGES_REQUESTED** [review]

**Comment by wcrzlh** [comment]
fixed

File: configs/cmt/cmt_small_ascend.yaml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+decay_epochs: 295
+warmup_epochs: 5
+
+# optimizer
+opt: "adamw"
+filter_bias_and_bn: True
+momentum: 0.9
+weight_decay: 0.05
+loss_scale_type: 'dynamic'
+loss_scale: 16777216.0
```
Comment (SamitHuang) [inline configs/cmt/cmt_small_ascend.yaml:59]: This is really large. How can you get this magic number?

Comment (wcrzlh) [reply configs/cmt/cmt_small_ascend.yaml:59]: I use the initial loss scale(2**24) in mindspore.amp.DynamicLossScaleManager. 
dynamiclossscale

**Review by SamitHuang — APPROVED** [review]

**Review by geniuspatrick — APPROVED** [review]

**Pull request merged**